web/src/components/core/ShowTerminalButton.test.jsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { screen } from "@testing-library/react";
+import { plainRender, mockComponent } from "~/test-utils";
+import { ShowTerminalButton } from "~/components/core";
+
+jest.mock("~/components/core/Terminal", () => mockComponent("Terminal Mock"));
+
+describe("ShowTerminalButton", () => {
+  it("renders a button for displaying logs", () => {
+    plainRender(<ShowTerminalButton />);
+    const button = screen.getByRole("button", "Terminal");
+    expect(button).not.toHaveAttribute("disabled");
+  });
+
+  describe("when user clicks on it", () => {
+    it("displays the Terminal component", async () => {
+      const { user } = plainRender(<ShowTerminalButton />);
+      const button = screen.getByRole("button", "Terminal");
+      await user.click(button);
+      screen.getByText(/Terminal Mock/);
+    });
+  });
```

Comment:
NP: for me, this test could be simplified by just using the second `it` and renaming it to something like `"renders a button that allows displaying the Terminal component"` or so. 

It's already testing that the button is rendered (it was found), that it is not disabled (click event didn't failed), and that it behaves as expected (the terminal mock was rendered) in just one shot.

I know that we used to write more granular test when using RSpec, but I feel that testing library is more about testing the use case.

Of course, I can be wrong. Either way, it is just a suggestion, your unit test is fine.